src/CMakeLists.txt

```diff
@@ -50,6 +50,24 @@ add_library(
     vulkan-renderer/world/indentation.cpp
 )
 
+foreach(FILE ${SOURCE_FILES})
+  get_filename_component(PARENT_DIR "${FILE}" PATH)
+
+  string(REPLACE "/" "\\" GROUP "${PARENT_DIR}")
+
+  if ("${FILE}" MATCHES ".*\\.cpp")
+    set(GROUP "Source Files\\${GROUP}")
+  elseif("${FILE}" MATCHES ".*\\.hpp")
+    set(GROUP "Header Files\\${GROUP}")
+  endif()
+
+  source_group("${GROUP}" FILES "${FILE}")
+endforeach()
+
+add_library(
+	${SOURCE_FILES}
+)
```

Comment:
```suggestion
add_library(${SOURCE_FILES})
```

Reply:
Oh right.
By the way we don't need `set_property(GLOBAL PROPERTY USE_FOLDERS ON)`.
Done.